test/integration/http2_upstream_integration_test.cc

```diff
@@ -306,16 +305,24 @@ TEST_P(Http2UpstreamIntegrationTest, UpstreamConnectionCloseWithManyStreams) {
                                                             {":authority", "host"}});
     encoders.push_back(&encoder_decoder.first);
     responses.push_back(std::move(encoder_decoder.second));
+
+    // Ensure we've established a connection before issuing the resets. This is necessary
```

Comment:
Couldn't this be lifted up above the for loop since this is HTTP/2  Basically, make an HTTP connection and then wait for the upstream connection, and then do the for loop? That might be simpler to understand?

Reply:
I think that won't work because we're waiting for an upstream connection, which won't be established until we send at least one request through to the router. Moving the check up causes the test to time out

Reply:
OK makes sense. Then can we do it just on the first iteration since we should just make a single connection and need to wait for it?